lib/compiler/src/substitution_logic.cc

```diff
@@ -76,7 +76,7 @@ struct EvaluateOperatorAttributeExpr {
     return evaluate_list_size(v);
   }
 private:
-  OperatorParameters attrs;
+  PCGOperatorAttrs attrs;
 };
 
 tl::optional<TensorAttributeValue> evaluate_tensor_attribute_expr(ParallelTensorShape const &, AttributeExpr<TensorAttributeKey> const &);
```

Comment:
`tl::` no longer necessary, just `#include "utils/optional.h"` and you can use plain `optional`